web/utils.py

```
@@ -414,6 +416,407 @@ def create_leaderboard_context(
+        membership.cancel_at_period_end = True
+        membership.save(update_fields=["cancel_at_period_end"])
+
+        # Record cancellation event
+        MembershipSubscriptionEvent.objects.create(
+            user=user,
+            membership=membership,
+            event_type="canceled",
+            data={"subscription_id": subscription.id, "canceled_at": timezone.now().isoformat()},
+        )
+
+        return {"success": True, "subscription": subscription}
```

Ruff (0.8.2)

Consider moving this statement to an `else` block

(TRY300)